src/server/tiering/small_bins.h

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <absl/container/flat_hash_map.h>
+
+#include <optional>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "server/tiering/disk_storage.h"
+
+namespace dfly::tiering {
+
+// Small bins accumulate small values into larger bins that fill up 4kb pages.
+// SIMPLEST VERSION for now.
+class SmallBins {
+ public:
+  using FilledBin = std::pair<unsigned /* id */, std::string>;
+
+  // List of locations of values for corresponding keys
+  using KeySegmentList = std::vector<std::pair<std::string /* key*/, DiskSegment>>;
```

Comment:
instead of commenting on basic types, I suggest  `using` to declare `Key`, `Id`.

Reply:
Also comment on Id that it's unique

Reply:
Will define Id. I suggest not to define Key = string anywhere because it's overkill, we don't to it anywhere. It's clear from the comments above, inline comments are just for readability